stripe/src/main/java/com/stripe/android/paymentsheet/repositories/StripeIntentRepository.kt

```diff
@@ -12,15 +13,15 @@ import kotlin.coroutines.CoroutineContext
 internal sealed class StripeIntentRepository {
     abstract suspend fun get(
         clientSecret: ClientSecret
-    ): PaymentIntent
+    ): StripeIntent
```

Comment:
We can talk this through, but you can make DefaultFlowController have a StripePaymentIntentRepostory and a StripeSetupIntentRepository, and pass this as a parameter to init.

This will really take advantage of the ClientSecret Abstraction you created (which I really like)!   Only the edges will know that there is anything different about them, and you will elliminate several if statements, thus reducing complexity and improving readability.